### PR TITLE
feat: new layout

### PR DESCRIPTION
**Pull request by rodrigopaixaorj**
For more informations about what we've done, please visit: https://github.com/Nottinghster/otclient/tree/new-layout

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,178 +1,292 @@
...
-- Idle Animation Support
-- Highlight Mouse Target (press shift to select any object)
-- Crosshair
-- Floor Shadowing
-- Floor View Mode (Normal, Fade, Locked, Always, Always with transparency)
-- Anti-Aliasing Mode Options (Note: Smooth Retro will consume a little more GPU)
-- Floating Effects Option
-- Optimized Terminal
-- Refactored Walk System
-- Support for more mouse buttons, for example 4 and 5
```
Comment (kojimhojo) [inline README.md]: readme  shouldn't replace the original.

File: data/setup.otml [hunk 2]
```diff
@@ -29,7 +29,7 @@ game
 
   render
     invisible-ticks-per-frame: 75
-    item-ticks-per-frame: 500
```
Comment (kojimhojo) [inline data/setup.otml:32]: What does this have to do with the new layout?

File: modules/corelib/http.lua [hunk 3]
```diff
@@ -181,8 +181,7 @@ function HTTP.onPost(operationId, url, err, data)
         err = err .. " (" .. data:sub(1, 100) .. ")"
       end
     end
-    -- data = result
-
+    data = result
```
Comment (kojimhojo) [inline modules/corelib/http.lua:184]: Where is this used?

File: modules/gamelib/game.lua [hunk 4]
```diff
@@ -58,7 +58,7 @@ function g_game.getSupportedClients()
         1012, 1013, 1020, 1021, 1022, 1030, 1031, 1032, 1033, 1034, 1035, 1036, 1037, 1038, 1039, 1040, 1041, 1050,
         1051, 1052, 1053, 1054, 1055, 1056, 1057, 1058, 1059, 1060, 1061, 1062, 1063, 1064, 1070, 1071, 1072, 1073,
         1074, 1075, 1076, 1080, 1081, 1082, 1090, 1091, 1092, 1093, 1094, 1095, 1096, 1097, 1098, 1099, 1100, 1281, 1285,
-        1286, 1287, 1291, 1300, 1314, 1316, 1320, 1321 }
+        1286, 1287, 1291, 1300, 1310, 1311, 1314, 1316, 1320, 1321, 1322, 1332 }
```
Comment (kojimhojo) [inline modules/gamelib/game.lua:61]: Are you sure that 1310 works on this client?

**Comment by kojimhojo** [comment]
What are the bugs // what are the missing features?

I think it would be a good idea to split

game mainpanel
in
- game minimap
- game_inventory
- game_mainpanel (healthMana + options)

so as not to confuse

**Comment by leo18df** [comment]
I have to issues do report thats happening with me, just show it in here?

**Comment by leo18df** [comment]
> I have two issues to report thats happening with me, just show it in here?


https://github.com/mehah/otclient/assets/28656626/57a30ef6-bf36-43fe-ba69-60c740e8e580

canary 13.32 main branch otcnewlayout 11x protocol

dead issue

**Comment by leo18df** [comment]
Visual screen flick happens all the time with rooftops shadows


https://github.com/mehah/otclient/assets/28656626/c0973b86-4d5c-4dba-9441-36409a3496cd

from any floor, it keeps repeating. there is no errors like that on otcv8, so its a issue related with otchnewlayout

**Comment by leo18df** [comment]
Last knowed issue after getting vocation 


https://github.com/mehah/otclient/assets/28656626/e50898af-4043-443c-b908-05f591ca9fb2

just remembering. There is none of those errors on otcv8

**Comment by rodrigopaixaorj** [comment]
> What are the bugs // what are the missing features?
> 
> I think it would be a good idea to split
> 
> game mainpanel in
> 
>     * game minimap
> 
>     * game_inventory
> 
>     * game_mainpanel (healthMana + options)
> 
> 
> so as not to confuse

Be free to help, you can do a commit

**Comment by kojimhojo** [comment]
@Nottinghster There’s an error in the tooltips; when an object is destroyed or hidden, the tooltip remains stuck with the mouse

**Comment by rodrigopaixaorj** [comment]
> @Nottinghster There’s an error in the tooltips; when an object is destroyed or hidden, the tooltip remains stuck with the mouse

It's a known error by me, it's in my to-do list to fix ;)

**Comment by kojimhojo** [comment]
@kokekanon you can help me ?

**Comment by kokekanon** [comment]
> @kokekanon you can help me ?

2414vv
in
https://github.com/Nottinghster/otclient/blob/new-layout/modules/corelib/ui/tooltip.lua#L62
add
```lua
local function onWidgetDestroy(widget)
    if widget == currentHoveredWidget then
        if widget.tooltip then
            g_tooltip.hide()
        end
        if widget.specialtooltip then
            g_tooltip.hideSpecial()
        end
        currentHoveredWidget = nil
    end
end

```

add 
`onDestroy = onWidgetDestroy,`
in
 https://github.com/Nottinghster/otclient/blob/new-layout/modules/corelib/ui/tooltip.lua#L119
https://github.com/Nottinghster/otclient/blob/new-layout/modules/corelib/ui/tooltip.lua#L148

**Comment by kojimhojo** [comment]
Obrigado

**Comment by kojimhojo** [comment]
> What are the bugs // what are the missing features?
> 
> I think it would be a good idea to split
> 
> game mainpanel in
> 
> * game minimap
> * game_inventory
> * game_mainpanel (healthMana + options)
> 
> so as not to confuse

image

**Comment by kojimhojo** [comment]
image

this does not work

**Comment by rodrigopaixaorj** [comment]
> image
> 
> this does not work

It's a known issue, but right now, don't have time to fix it
You can help us with this fix

**Review by majestyotbr — APPROVED** [review]
